src/high-level/papi_hl.c

```diff
@@ -1055,20 +1075,19 @@ static int _internal_hl_read_counters()
 {
    int i, j, retval;
    for ( i = 0; i < num_of_components; i++ ) {
-      if ( i < ( num_of_components - 1 ) ) {
-         retval = PAPI_read( _local_components[i].EventSet, _local_components[i].values);
-      } else {
-         /* get cycles for last component */
-         retval = PAPI_read_ts( _local_components[i].EventSet, _local_components[i].values, &_local_cycles );
-      }
+      retval = PAPI_read( _local_components[i].EventSet, _local_components[i].values);
```

Comment:
Q: In the PR description you mentioned that "internal_hl_read_counters that wasn't reading the counters for all components in the machines we tested". What exactly was your setup for this i.e. PAPI configure, system you were on, exported `PAPI_EVENTS`, etc.

I used the master `papi_hl.c` with this branch for the three new nvml events you added and all three show with values in the .json I end up creating.